ui/src/main/java/com/dat3m/ui/options/OptionsPane.java

```diff
@@ -73,34 +86,38 @@ public OptionsPane(){
         mkGrid();
     }
 
-    private void bindListeners(){
-		// optionsPane needs to listen to options to clean the console
-		// Alias and Mode do not change the result and thus we don't listen to them 
-		targetPane.addActionListener(this);
-		boundField.addActionListener(this);
-		timeoutField.addActionListener(this);
-		clearButton.addActionListener(this);
+    private void bindListeners() {
+        // optionsPane needs to listen to options to clean the console
+        // Alias and Mode do not change the result, and thus we don't listen to them
+        targetPane.addActionListener(this);
+        boundField.addActionListener(this);
+        timeoutField.addActionListener(this);
+        clearButton.addActionListener(this);
+        propertyPane.addActionListener(this);
     }
 
-    public JButton getTestButton(){
+    public JButton getTestButton() {
         return testButton;
     }
 
-    public JTextPane getConsolePane(){
+    public JTextPane getConsolePane() {
         return consolePane;
     }
 
-    public UiOptions getOptions(){
+    public UiOptions getOptions() {
         int bound = Integer.parseInt(boundField.getText());
         int timeout = Integer.parseInt(timeoutField.getText());
-        String cflags = cflagsField.getText();
-        Arch target = (Arch)targetPane.getSelectedItem();
-        Method method = (Method)methodPane.getSelectedItem();
-        Solvers solver = (Solvers)solverPane.getSelectedItem();
-        return new UiOptions(target, method, bound, solver, timeout, cflags);
+        boolean showViolationGraph = showViolationField.isSelected();
+        String cflags = cflagsField.getText().strip();
+        String config = configField.getText().strip();
+        Arch target = (Arch) targetPane.getSelectedItem();
+        Method method = (Method) methodPane.getSelectedItem();
+        Solvers solver = (Solvers) solverPane.getSelectedItem();
+        EnumSet<Property> properties = EnumSet.of((Property) propertyPane.getSelectedItem());
```

Comment:
Since we just have a few properties (TBH, I would filter out the SVCOMP data races), wouldn't it makes more sense to have this as a button (similar to showViolationGraph) rather than a selector? This way the user can check more than one property at a time.

Reply:
I think there are plenty of things you could do with the UI, especially when considering user-friendliness.
The features I added in this PR are mostly related to doing live demonstrations (like enlarging fonts/zooming), i.e., the user is me.

Radio buttons come with the problem that not a single one may be selected, so you need error reporting as well.
Similarly, multiple properties fail for litmus code in many case so you also need error reporting there (I guess the exception error in the console might be enough).
None of that is particularly hard, but should be done in a different PR IMO.